src/System.Text.CaseFolding/System/Text/CaseFolding/SimpleCaseFolding.cs

```diff
@@ -0,0 +1,630 @@
+                    dst = ref Unsafe.Add(ref dst, 1);
+                    length--;
+                }
+
+                if (length == 0)
+                {
+                    return;
+                }
+
+                // We catch a char above 0x5ff.
+                // Process it with more slow two-level mapping.
+                while (length != 0 && !IsSurrogate(src))
+                {
+                    var v1 =  Unsafe.Add(ref refMapLevel1, src >> 8);
+                    var ch1 = Unsafe.Add(ref refMapData, v1 + (src & 0xFF));
+                    if (ch1 == 0)
+                    {
+                        ch1 = src;
+                    }
+
+                    dst = ch1;
+                    src = ref Unsafe.Add(ref src, 1);
+                    dst = ref Unsafe.Add(ref dst, 1);
+                    length--;
+                }
+
+                if (length == 0)
+                {
+                    return;
+                }
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool IsAscii(char c)
+        {
+            return c < 0x80;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool IsHighSurrogate(char c)
+        {
+            return (uint)(c - HIGH_SURROGATE_START) <= (uint)(HIGH_SURROGATE_END - HIGH_SURROGATE_START);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool IsLowSurrogate(char c)
+        {
+            return (uint)(c - LOW_SURROGATE_START) <= (uint)(LOW_SURROGATE_END - LOW_SURROGATE_START);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool IsSurrogate(char c)
+        {
+            return (uint)(c - HIGH_SURROGATE_START) <= (uint)(LOW_SURROGATE_END - HIGH_SURROGATE_START);
+        }
+
+        /// <summary>
+        /// Search the char position in the string with simple case folding.
+        /// </summary>
+        /// <param name="source">Source string.</param>
+        /// <param name="ch">Char to search.</param>
+        /// <returns>
+        /// Returns an index the char in the string or -1 if not found.
+        /// </returns>
+        public static int IndexOfFolded(this string source, char ch)
+        {
+            if (source == null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+
+            return IndexOfFolded(source.AsSpan(), ch);
+        }
+
+        /// <summary>
+        /// Search the char position in the ReadOnlySpan&lt;char&gt; with simple case folding.
+        /// </summary>
+        /// <param name="source">Source string.</param>
+        /// <param name="ch">Char to search.</param>
+        /// <returns>
+        /// Returns an index the char in the ReadOnlySpan&lt;char&gt; or -1 if not found.
+        /// </returns>
+        public static int IndexOfFolded(this ReadOnlySpan<char> source, char ch)
```

Comment:
Simple Case Folding is aimed at identifiers mainly so typical length is less 64.
On the other hand, it can/should be used in searching (and regex) as a replacement for OrdinalIgnoreCase - search patterns could be more 64.